.kokoro/tests/run_tests.sh

```diff
@@ -103,109 +93,24 @@ if [[ ",$JAVA_VERSION," =~ "11" ]]; then
-    cd "$ROOT"
-    # Navigate to the project folder.
-    file=$(dirname "$file")
-    cd "$file"
-
-    # Skip Legacy GAE
-    if [[ "$SKIP_LEGACY_GAE" = "true" ]]; then
-      if [[ ",$JAVA_VERSION," =~ "11" ]]; then
-        case "$file" in
-          *appengine-java8*)
-            continue
-            ;;
-          *flexible*)
-            continue
-            ;;
-        esac
-      fi
-    fi
-
-    # If $DIFF_ONLY is true, skip projects without changes.
-    if [[ "$ONLY_DIFF" = "true" ]]; then
-        git diff --quiet origin/master.. .
-        CHANGED=$?
-        if [[ "$CHANGED" -eq 0 ]]; then
-          # echo -e "\n Skipping $file: no changes in folder.\n"
-          continue
-        fi
-    fi
-
-    echo "------------------------------------------------------------"
-    echo "- testing $file"
-    echo "------------------------------------------------------------"
-
-    date
-    
-    # Fail the tests if no Java version was found.
-    POM_JAVA=$(grep -oP '(?<=<maven.compiler.target>).*?(?=</maven.compiler.target>)' pom.xml)
-    ALLOWED_VERSIONS=("1.8" "11")
-    # shellcheck disable=SC2199
-    # shellcheck disable=SC2076
-    if [[ "$POM_JAVA" = "" ]] || [[ !  "${ALLOWED_VERSIONS[@]}" =~ "${POM_JAVA}" ]]; then
-        RTN=1
-        echo -e "\n Testing failed: Unable to determine Java version. Please set in pom:"
-        echo -e "\n<properties>"
-        echo -e "  <maven.compiler.target>1.8</maven.compiler.target>"
-        echo -e "  <maven.compiler.source>1.8</maven.compiler.source>"
-        echo -e "</properties>\n"
-        continue
-    fi
-
-    # Skip tests that don't have the correct Java version.
-    # shellcheck disable=SC2076
-    if ! [[ ",$JAVA_VERSION," =~ ",$POM_JAVA," ]]; then
-        echo -e "\n Skipping tests: Java version ($POM_JAVA) not required ($JAVA_VERSION)\n"
-        continue
-    fi
-
-    # Use maven to execute the tests for the project.
-    mvn --quiet --batch-mode --fail-at-end clean verify \
-       -Dfile.encoding="UTF-8" \
-       -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
-       -Dmaven.test.redirectTestOutputToFile=true \
-       -Dbigtable.projectID="${GOOGLE_CLOUD_PROJECT}" \
-       -Dbigtable.instanceID=instance
-    EXIT=$?
-
-    if [[ $EXIT -ne 0 ]]; then
-      RTN=1
-      echo -e "\n Testing failed: Maven returned a non-zero exit code. \n"
-    else
-      echo -e "\n Testing completed.\n"
-    fi
-
-    # Build and deploy Cloud Run samples
-    if [[ "$file" == "run/"* ]]; then
-      export SAMPLE_NAME=${file#"run/"}
-      # chmod 755 "$SCRIPT_DIR"/build_cloud_run.sh
-      "$SCRIPT_DIR"/build_cloud_run.sh
-      EXIT=$?
-
-      if [[ $EXIT -ne 0 ]]; then
-        RTN=1
-        echo -e "\n Cloud Run build/deploy failed: gcloud returned a non-zero exit code. \n"
-      else
-        echo -e "\n Cloud Run build/deploy completed.\n"
-      fi
-    fi
+btlr_args=(
+    "run"
+    "--max-cmd-duration=20m"
```

Comment:
I wonder if this is true.

Reply:
this is a limitation enforced by btlr - it'll *should* kill the process once this time-limit expires. I left this deliberately low for now to try and help spot troublesome projects that are taking too long to run tests on.